vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -0,0 +1,418 @@
+        tuning_data : pd.DataFrame or str, default = None
+            Another dataset containing validation data reserved for model selection and hyperparameter-tuning,
+            can be a dataframe like image dataset.
+            If a string is provided, will search for k8 datasets.
+            If `None`, the validation dataset will be randomly split from `train_data` according to `holdout_frac`.
+        time_limit : int, default = None
+            Time limit in seconds, if not specified, will run until all tuning and training finished.
+            If `time_limit` is hit during `fit`, the
+            HPO process will interrupt and return the current best configuration.
+        hyperparameters : dict, default = None
+            Extra hyperparameters for specific models.
+            Accepted args includes(not limited to):
+            epochs : int, default value based on network
+                The `epochs` for model training.
+            net : mx.gluon.Block
+                The custom network. If defined, the model name in config will be ignored so your
+                custom network will be used for training rather than pulling it from model zoo.
+            optimizer : mx.Optimizer
+                The custom optimizer object. If defined, the optimizer will be ignored in config but this
+                object will be used in training instead.
+            batch_size : int
+                Mini batch size
+            lr : float
+                Trainer learning rate for optimization process.
+            You can get the list of accepted hyperparameters in `config.yaml` saved by this predictor.
+        **kwargs :
+            holdout_frac : float, default = 0.1
+                The random split ratio for `tuning_data` if `tuning_data==None`.
+            random_state : int, default = None
+                The random_state(seed) for shuffling data, only used if `tuning_data==None`.
+                Note that the `random_state` only affect the splitting process, not model training.
+                If not specified(None), will leave the original random sampling intact.
+            nthreads_per_trial : int, default = (# cpu cores)
+                Number of CPU threads for each trial, if `None`, will detect the # cores on current instance.
+            ngpus_per_trial : int, default = (# gpus)
+                Number of GPUs to use for each trial, if `None`, will detect the # gpus on current instance.
+            hyperparameter_tune_kwargs: dict, default = None
+                num_trials : int, default = 1
+                    The number of HPO trials when `time_limit` is None. 
+                search_strategy : str, default = 'random'
+                    Searcher strategy for HPO, 'random' by default.
+                    Options include: ‘random’ (random search), ‘bayesopt’ (Gaussian process Bayesian optimization),
+                    ‘grid’ (grid search).
+                max_reward : float, default = 0.95
```

Comment:
I feel that this should default to 1 or effectively not be set. I think there is little reason to stop HPO early when achieving 95% accuracy, especially for datasets that are imbalanced and it is trivial to get 95% accuracy without much meaningful benefits.

Reply:
Another reason for not setting this at all would be if in future we had `log_loss` as an evaluation metric. A max_reward of 0.95 wouldn't make sense in this case.

Reply:
Agreed I think this should default to infinity and we should not assume metrics will always lie in [0,1].